torch/utils/data/dataloader.py

```diff
@@ -174,6 +190,15 @@ def __init__(self, loader):
     def __len__(self):
         return len(self.batch_sampler)
 
+    def _get_batch(self):
+        if self.timeout > 0:
+            signal.alarm(self.timeout)
+            result = self.data_queue.get()
+            signal.alarm(0)
```

Comment:
I don't really like `SIGALRM`, because it's a global thing. We might run into problems when we'll have multiple `DataLoaders` in a single process